lstchain/reco/dl0_to_dl1.py

```diff
@@ -312,37 +314,35 @@ def r0_to_dl1(input_filename = get_dataset_path('gamma_test_large.simtel.gz'),
                     dl1_container.gps_time = event.trig.gps_time.value
 
                     if not is_simu:
-                        # GPS + WRS + UCTS is now working in its nominal configuration. 
+                        # GPS + WRS + UCTS is now working in its nominal configuration.
                         # These TS are stored into ucts_time container.
-                        # TS can be alternatively calculated from the TIB/Dragon 
-                        # modules counters + NTP time corresponding to the start 
-                        # of the run (just for cross-checks). For the time being,
-                        # the three TS will be stored in the DL1 files.
-                        # This will be deprecated and modified back to uniquely use
-                        # gps_time whenever the GPS + WRS + UCTS info reliably and stably
-                        # arrives at the EvB side.
+                        # TS can be alternatively calculated from the TIB and 
+                        # Dragon modules counters based on the first valid UCTS TS
+                        # as the reference point. For the time being, the three TS
+                        # are stored in the DL1 files for checking purposes.
 
-                        # gps_time = event.r0.tel[telescope_id].trigger_time
+                        ucts_time = event.lst.tel[telescope_id].evt.ucts_timestamp * 1e-9  # secs
 
-                        ucts_time = event.lst.tel[telescope_id].evt.ucts_timestamp * 1e-9 # nsecs
+                        module_id = 82  # Get counters from the central Dragon module
 
-                        # Get counters from the central Dragon module
-                        module_id = 82
-
-                        dragon_time = (
-                                event.lst.tel[telescope_id].svc.date +
+                        if ucts_t0 and dragon_counter0:
+                            dragon_time = ((ucts_t0 - dragon_counter0) * 1e-9 +  # secs
+                                event.lst.tel[telescope_id].evt.pps_counter[module_id] +
+                                event.lst.tel[telescope_id].evt.tenMHz_counter[module_id] * 10**(-7))
```

Comment:
at the moment, no `tib_time` if `ucts_t0` and `dragon_counter0` are given

Reply:
Right now yes. Isidro already provided me the values to calculate TIB TS so I will implement it.

```
 01915    4 DRS4  2020-01-31 21:33:34 0001 1580506462208626706 2208626500 0001 1580506462208626706 2208626500  
 01916    5 CALI  2020-01-31 22:09:15 0001 1580508610052979006 9052978700 0001 1580508610052979006 9052978600  
 01917    4 CALI  2020-02-01 01:00:32 0001 1580518883322614305 5322614000 0001 1580518883322614305 5322614000  
 01918  271 DATA  2020-02-01 01:12:35 0001 1580519604210915532 4210915300 0001 1580519604210915532 4210915400  
 01919  299 DATA  2020-02-01 01:44:05 0003 1580521498164948130 6164947900 0003 1580521498164948130 6164947800  
 01920  296 DATA  2020-02-01 02:19:23 0001 1580523618574809957 8574809700 0001 1580523618574809957 8574809800  
 01921  342 DATA  2020-02-01 02:54:34 0001 1580525726240811910 5240811700 0001 1580525726240811910 5240811800 
```

 The drawback here is that I need to add two new extra `ucts_t0_tib` and `tib_counter0` variables since we need to find an event (ucts_t0_tib and its corresponding counter) with valid UCTS and TIB flag which does not have to be necessarily the same event used to build Dragon TS. Do you see a problem here having so many options?

Reply:
ok, since this is temporary...

Reply:
implemented the calculation of TIB timestamp base on the first valid TIB counter and ucts_time